### DISCUSSION TITLE
Build a java client by maven or gradle

### DISCUSSION THREAD

**Discussion by jainvikas8** [g0hl1n/meta-java#28] [General]
Hi,

 Are there any `classes/recipe/layers` that could be used to build a provider/java client and the resulting jar file can be used by [amazon-corretto](https://github.com/aws4embeddedlinux/meta-aws/tree/master/recipes-devtools/amazon-corretto) for Greengrass which can be integrated as part of distro image?

There is maven and gradle, but only can find https://github.com/intel-iot-devkit/meta-iot-cloud/blob/master/recipes-support/maven/maven_3.6.3.bb which seems to be old.

   Any thoughts, please?

Thanks.

**Comment by moto-timo** [comment]
Modern usage of `maven` and `gradle` will recall newer recipes to be added to this layer. Maven itself is a problem because it must be bootstrapped by an N-1 version. I have no experience with attempting to build `gradle` from source.

For Greengrass and amazon-corretto, you are probably better off discussing with folks from the  [meta-aws](https://github.com/aws4embeddedlinux/meta-aws) layer. If there is interest in moving forward with such work, we welcome input from the community, but it must be clear that it will require contributions from those that need it.
